api/src/main/java/org/openmrs/module/kenyaemr/api/impl/KenyaEmrServiceImpl.java

```diff
@@ -167,6 +170,24 @@ public String getNextHivUniquePatientNumber(String comment) {
 		return prefix+sequentialNumber;
 	}
 
+	/**
+	 * @see org.openmrs.module.kenyaemr.api.KenyaEmrService#getNextPrepUniquePatientNumber(String)
+	 */
+	@Override
+	public String getNextPrepUniquePatientNumber(String comment) {
+		if (comment == null) {
+			comment = "KenyaEMR Service";
+		}
+
+		PatientIdentifierType upnType = MetadataUtils.existing(PatientIdentifierType.class, HivMetadata._PatientIdentifierType.UNIQUE_PATIENT_NUMBER);
```

Comment:
this may not be for you, but is this the CCC Number or something? The method is for generating prep number. please clarify

Reply:
`java.lang.NullPointerException\r\n\tat org.openmrs.module.idgen.service.BaseIdentifierSourceService.generateIdentifiers(BaseIdentifierSourceService.java:172)\r\n\tat org.openmrs.module.idgen.service.BaseIdentifierSourceService.generateIdentifier(BaseIdentifierSourceService.java:254)`

This is the error i get if i switch to prep identifier uuid